code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/identity/EnsureSettingForProvider.java

```diff
@@ -0,0 +1,75 @@
+package io.cattle.platform.iaas.api.auth.identity;
+
+import io.cattle.platform.api.auth.Identity;
+import io.cattle.platform.archaius.util.ArchaiusUtil;
+import io.cattle.platform.iaas.api.auth.SecurityConstants;
+import io.cattle.platform.iaas.api.auth.SettingsUtils;
+import io.cattle.platform.iaas.api.auth.integration.github.GithubConstants;
+import io.cattle.platform.iaas.api.auth.integration.github.GithubUtils;
+import io.cattle.platform.util.type.InitializationTask;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import javax.inject.Inject;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+public class EnsureSettingForProvider implements InitializationTask {
+
+    private static final Log logger = LogFactory.getLog(EnsureSettingForProvider.class);
+
+    @Inject
+    SettingsUtils settingsUtils;
+
+    @Inject
+    GithubUtils githubUtils;
+
+    @Override
+    public void start(){
+        if (SecurityConstants.SECURITY.get() &&
+                StringUtils.isNotBlank(GithubConstants.GITHUB_CLIENT_SECRET.get()) &&
+                StringUtils.isNotBlank(GithubConstants.GITHUB_CLIENT_ID.get())){
+            if (StringUtils.isBlank(SecurityConstants.AUTH_PROVIDER.get())) {
+                settingsUtils.changeSetting(SecurityConstants.AUTH_PROVIDER_SETTING, GithubConstants.CONFIG);
+                logger.info("Upgrade detected. Adding setting for :" + SecurityConstants
+                        .AUTH_PROVIDER_SETTING);
+
+            }
+            if ("restricted".equalsIgnoreCase(GithubConstants.ACCESS_MODE.get()) &&
+                    StringUtils.isBlank(GithubConstants.GITHUB_ALLOWED_IDENTITIES.get())){
```

Comment:
It means for the setups having this setting set, the migration won't be performed? Would it cause any problems? We should at least document it if it does